Firestore/core/src/firebase/firestore/nanopb/message.h

```diff
@@ -55,68 +47,101 @@ using MaybeMessage = util::StatusOr<Message<T>>;
 /**
  * A unique-ownership RAII wrapper for Nanopb-generated message types.
  *
- * Nanopb-generated message types are plain C structs that contain some
- * dynamically-allocated memory and should be deallocated by calling
- * `pb_release`; `Message` implements a simple RAII wrapper that does just that.
- * For simplicity, `Message` implements unique ownership and is immutable after
- * construction (not counting friend classes). Use `proto()` member function to
- * access the underlying proto.
- *
- * `Message` provides a pointer-like access to the underlying Nanopb-generated
- * message type.
+ * Nanopb-generated message types (from now on, "Nanopb protos") are plain
+ * C structs that contain some dynamically-allocated memory and should be
+ * deallocated by calling `pb_release`; `Message` implements a simple RAII
+ * wrapper that does just that. For simplicity, `Message` implements unique
+ * ownership model. It provides a pointer-like access to the underlying Nanopb
+ * proto. Also, `Message` serves to translate representation formats between
+ * Nanopb and gRPC.
  *
  * Note that moving *isn't* a particularly cheap operation in the general case.
- * Even without doing deep copies, Nanopb-generated messages may contain *a lot*
- * of member variables.
+ * Even without doing deep copies, Nanopb protos contain *a lot* of member
+ * variables (at the time of writing, the largest `sizeof` of a Nanopb proto was
+ * 248).
  */
 template <typename T>
 class Message {
```

Comment:
Design of `Message` has changed quite a bit. It no longer even tries to pretend to be immutable.